### DISCUSSION TITLE
How to get every valid field after a blur?

### DISCUSSION THREAD

**Discussion by lucasmaffazioli** [react-hook-form/react-hook-form#7613] [Q&A] [answered]
I have multiple fields that act as filters, I want that after every field blur I call a function with the current valid values of the form. I also don't need a submit after completing the "form".

I've tried an `useEffect` with the `getValues()` or `watch` as condition to get the current values, but it doesn't work.

**Comment by Moshyfawn** [comment, marked as answer]
You can try form's `onBlur` event. See [this CSB](https://codesandbox.io/s/onblur-submit-example-wii4j)

**Reply by lucasmaffazioli** [reply]
This one I already saw, and like I've said I want to do it after every field blur, not after the form blur.

**Reply by Moshyfawn** [reply]
I can't see how the form `onBlur` approach is different from your requirement, tbh.
The submit happens exactly on field blur, nothing else

https://user-images.githubusercontent.com/16290753/150777307-524092bd-c61d-49f4-87cb-ec6baa8c074b.mp4

**Reply by lucasmaffazioli** [reply]
The submit count increases after the first blur, but how do I get the value at this first blur?

**Reply by Moshyfawn** [reply]
What do you mean "at this first blur"? If you pass the submit handler into the form's `onBlur` event
```js
<form onBlur={handleSubmit(onSubmit)} />
```
the form validation and the submit callback after will fire _(provided no validation errors occurred)_. Form values are passed to `onSubmit` from `handleSubmit` this way.

On the screencast above the submitted values don't change after the first field value provided and blured due to the Age field being empty, tho, the field is set as required.
```js
{...register("age", {
  required: "Age field is required"
})}
```
So, the submit count is increased, but it's a failed submit due to the form validation error occurred in `handleSubmit`; you see Submitted Data doesn't change because of it

**Reply by lucasmaffazioli** [reply]
Yes it was the required, without it my requirements are all met! Thank you!
